src/controllers/Chat.controller.ts

```diff
@@ -1,25 +1,41 @@
-import { Socket } from "socket.io";
+import { Socket, Server } from "socket.io";
 import Chat from "../database/models/Chat.model";
 import { Request, Response } from "express";
+import { or } from "sequelize";
 
-export const chat = (socket: Socket) => {
-  socket.on("chat message", async (msg) => {
-    const { email, surname, given_name, id } = socket.data.user;
-    Chat.create({ message: msg, senderId: id }).then(() => {
-      socket.broadcast.emit("chat message", {
-        message: msg,
-        sender: { email, surname, given_name },
-      });
+const rooms = new Map();
+
+export function chat(socket: Socket, io: Server) {
+  socket.on("room", ({ room, to }) => {
+    const receiver = rooms.get(to);
+    if (!receiver) {
+      socket.join(room);
+      rooms.set(socket.data.user.id, room);
+    }
+    socket.on("chat message", async (msg) => {
+      const { email, surname, given_name, id } = socket.data.user;
+      Chat.create({ message: msg, senderId: id, receiverId: to })
+        .then(() => {
+          io.sockets.in(receiver || room).emit("chat message", {
+            message: msg,
+            sender: { email, surname, given_name },
+          });
+        })
+        .catch((err) => {
+          console.log(err.message);
```

Comment:
```suggestion
          console.log(err.message);
```
can you just if is possible send message to the FE in case of error

.catch(() => {
          socket.emit("error", "An error occurred while sending the message.");
        });